### DISCUSSION TITLE
Switching Command and Option on Mac

### DISCUSSION THREAD

**Discussion by skanev** [wezterm/wezterm#2754] [General]
Hey! WezTerm is really amazing and I love it. I've been using it on Linux for a while, and it's really great.

I've been trying to move to it on Mac (from iTerm2), but I ran into an incompatibility with my workflow. Specifically, I'd like to be able to switch Command and Option. Basically, I need Command to function as meta, so I can use alt+b/alt+f or cmd+b/cmd+f to move backward/forward a word in zsh (I do it this way because it's at the same place on the keyboard, immediately left of the space bar, and therefore muscle memory works).

I've been going through the docs to figure out how to do it, but I can't find a setting. Am I missing anything?

**Comment by wez** [comment]
Take a look at:
* https://github.com/wez/wezterm/issues/1777

There's no built-in function in wezterm, and I probably won't add it, but there is a recommendation for other software that you can install on your mac to gain a very high degree of control over your keyboard.

**Reply by skanev** [reply]
Fair enough. I appreciate the complexity of doing a full-blown keyboard remapping exercise, and maybe using-cmd-as-meta-in-the-terminal is not as common as I think it is (which is what drove me to iTerm in the first place).

I ended up writing a bunch of lua here:

```lua
local wezterm = require('wezterm')
local environment = require('environment')

local all_characters = [[`1234567890-=qwertyuiop[]\asdfghjkl;'zxcvbnm,./]]
local characters = {}

for i = 1, #all_characters do
  table.insert(characters, all_characters:sub(i, i))
end

local function mappings(maps)
  local result = {}
  local seen = {}
  local mod_key

  if environment.os == 'mac' then
    mod_key = 'CMD'
  else
    mod_key = 'ALT'
  end

  for _, mapping in ipairs(maps) do
    mapping.mods = mapping.mods and mapping.mods:gsub('MOD', mod_key)
    table.insert(result, mapping)

    seen[mapping.mods .. " " .. mapping.key:lower()] = true
  end

  if mod_key == 'CMD' then
    for _, key in ipairs(characters) do
      for _, mods in ipairs({ 'CMD', 'CMD|SHIFT' }) do
        local combo = mods .. ' ' .. key

        if not seen[combo] then
          seen[combo] = true

          table.insert(result, {
            key = key,
            mods = mods,
            action = wezterm.action.SendKey { key = key, mods = mods:gsub('CMD', 'ALT') },
          })
        end
      end
    end
  end

  return result
end

return {
  keys = mappings
}
```

The interesting bit is the second half of `mappings` which basically goes and defines CMD-A to send ALT-A and CMD-SHIFT-Z to send ALT-SHIFT-Z and so on. The rest of it is more relevant to my setup (I'd like to say `MOD` instead of `CMD`, so it can mean `CMD` on mac and `ALT` on linux/windows, and since I'm passing all those mappings through this function, it makes sure I don't clobber something else). Haven't used it in full anger yet, but seems to do the trick for now.
